### PR TITLE
refactor: remove all usages of v_init and deprecate

### PR DESCRIPTION
**Pull request by jackgerrits**

### CODE REVIEW

**Review by lalo — APPROVED** [review]

File: vowpalwabbit/v_array.h [hunk 1]
```diff
@@ -415,6 +415,7 @@ struct v_array<T, typename std::enable_if<std::is_trivially_copyable<T>::value>:
 };
 
 template <class T>
+VW_DEPRECATED("Use v_array's constructor directly instead. This will be removed in VW 9.0.")
 inline v_array<T> v_init()
```
Comment (lalo) [inline vowpalwabbit/v_array.h:419]: if we were to delete it today with this PR, are all the references in this repo to it gone?

File: vowpalwabbit/conditional_contextual_bandit.cc [hunk 2]
```diff
@@ -188,7 +188,7 @@ void save_action_scores(ccb& data, decision_scores_t& decision_scores)
   data.exclude_list[original_index_of_chosen_action] = true;
 
   decision_scores.emplace_back(std::move(pred));
-  data.shared->pred.a_s = v_init<ACTION_SCORE::action_score>();
+  data.shared->pred.a_s = v_array<ACTION_SCORE::action_score>();
```
Comment (peterychang) [inline vowpalwabbit/conditional_contextual_bandit.cc:191]: would `clear()` work? I think it should

File: vowpalwabbit/v_array.h (hunk 1 above)
Comment (jackgerrits) [reply vowpalwabbit/v_array.h:419]: yes

**Review by peterychang — APPROVED** [review]
I would recomment using `clear()` to get moved v_arrays into a known state, but otherwise looks good!

**Pull request merged**